dspace-api/src/main/java/org/dspace/external/provider/impl/OrcidV2AuthorDataProvider.java

```diff
@@ -0,0 +1,280 @@
+                        try {
+                            responseObject = new JSONObject(inputStr);
+                        } catch (Exception e) {
+                            //Not as valid as I'd hoped, move along
+                            responseObject = null;
+                        }
+                    }
+                }
+            }
+            if (responseObject != null && responseObject.has("access_token")) {
+                accessToken = (String) responseObject.get("access_token");
+            }
+        }
+    }
+
+    /**
+     * Makes an instance of the Orcidv2 class based on the provided parameters.
+     * This constructor is called through the spring bean initialization
+     */
+    private OrcidV2AuthorDataProvider(String url) {
+        this.orcidRestConnector = new OrcidRestConnector(url);
+    }
+
+    @Override
+    public Optional<ExternalDataObject> getExternalDataObject(String id) {
+        Person person = getBio(id);
+        ExternalDataObject externalDataObject = convertToExternalDataObject(person);
+        return Optional.of(externalDataObject);
+    }
+
+    protected ExternalDataObject convertToExternalDataObject(Person person) {
+        ExternalDataObject externalDataObject = new ExternalDataObject(sourceIdentifier);
+        String lastName = "";
+        String firstName = "";
+        if (person.getName().getFamilyName() != null) {
+            lastName = person.getName().getFamilyName().getValue();
+            externalDataObject.addMetadata(new MockMetadataValue("person", "familyName", null, null,
+                                                                 lastName));
+        }
+        if (person.getName().getGivenNames() != null) {
+            firstName = person.getName().getGivenNames().getValue();
+            externalDataObject.addMetadata(new MockMetadataValue("person", "givenName", null, null,
+                                                                 firstName));
+
+        }
+        externalDataObject.setId(person.getName().getPath());
+        externalDataObject
+            .addMetadata(new MockMetadataValue("dc", "identifier", "orcid", null, person.getName().getPath()));
+        externalDataObject
+            .addMetadata(new MockMetadataValue("dc", "identifier", "uri", null, orcidUrl + person.getName().getPath()));
+        if (!StringUtils.isBlank(lastName) && !StringUtils.isBlank(firstName)) {
+            externalDataObject.setDisplayValue(lastName + ", " + firstName);
+            externalDataObject.setValue(lastName + ", " + firstName);
+        } else if (StringUtils.isBlank(firstName)) {
+            externalDataObject.setDisplayValue(lastName);
+            externalDataObject.setValue(lastName);
+        } else if (StringUtils.isBlank(lastName)) {
+            externalDataObject.setDisplayValue(firstName);
+            externalDataObject.setValue(firstName);
+        }
+        return externalDataObject;
```

Comment:
I think this way of mapping Orcid, in the code, is less flexible and might be limiting access to other relevant information like the person's identifiers (scopus, researcher id, etc...), or author's name variants. I think XSL transformations could be used here, with the output of DSpace Intermediate Metadata: https://wiki.duraspace.org/display/DSPACE/DSpaceIntermediateMetadata

Reply:
If relevant, it can be addressed in another Pull Request

Reply:
That's a good suggestion, but not really the scope of this PR. The goal here is to have the framework in place and a sample implementation.
This can best be handled in a separate ticket